src/util/component-types.ts

```diff
@@ -304,26 +334,26 @@ export function getFirst<P>(
 //
 // Note: The Partial<P> type is referring to the props passed into the omitProps,
 // not the props defined on the component.
-export function omitProps<P extends object>(
-	props: Partial<P>,
-	component: ICreateClassComponentClass<P> | undefined,
-	keys: string[] = [],
-	targetIsDOMElement = true
-): { [key: string]: any } {
-	// We only want to exclude the `callbackId` key when we're omitting props
-	// destined for a dom element.
-	// We always want to exclude the `initialState` key.
-	const additionalOmittedKeys = targetIsDOMElement
-		? ['initialState', 'callbackId']
-		: ['initialState'];
-
-	// this is to support non-createClass components that we've converted to TypeScript
-	if (component === undefined) {
-		return _.omit(props, keys.concat(additionalOmittedKeys));
-	}
+// export function omitProps<P extends object>(
```

Comment:
Hey @yasskin did you mean to leave this commented out or did you mean to remove it?

Reply:
Lines 318 to 325 give the explanation.